### ISSUE TITLE
Question: Why ARP and multicast from discarding port

### ISSUE THREAD

**Issue by hjelmeland** [mstpd/mstpd#24]
Maybe this is more a Linux kernel question, but I start by asking you gentlemen for help.

We have a linux device with two ports running mstpd in RSTP mode. The idea is to provide redundant network connections to two interconnected switches. The device runs applications that tries to make connections. What we see is that ARP requests are sent on both interfaces, even when the interface is in "discarding" state. This seem to confuse the forwarding tables of the central switches. Same with multicasts originating from device. I tried "ifconfig <dev> -arp", but it did not help.  It is something else that can be configured to avoid this, or is it an inherit limitation of Linux bridging?

**Comment by commodo** [comment]
Hey ; so, for my use case, we use mstpd on top of switch ASICs (so, switch ASICs acting as bridges).

I must admit, I did not check what the ASIC does, with respect to broadcast and multicast send.
Looking through the bridge STP kernel code, I found this comment in` br_stp.c`:
```
                /* Multicast is not disabled for the port when it goes in
                 * blocking state because the timers will expire and stop by
                 * themselves without sending more queries.
                 */
```

Maybe setting some flags on the bridge port interfaces would be required.
There are some BR_FLOOD flags that can be/unset set via sysfs per port.
Maybe we should check it out ; I can try to look over this a bit more in-depth on Monday.

Thanks
Alex

**Comment by hjelmeland** [comment]
It turned out I fooled my self. On closer inspection I find no ARP or multicasts from blocked ports. (I was fooled because wireshark does no show direction, and I had bridged the interfaces on the PC that I used for tracing...) Very sorry about the noise. 

But what was a real issue was that we want to turn on "bridge assurance" / setportnetwork. Then the alternate core switch become confused because the blocked port sends BPDUs. I think the solution is to set different mac addresses on the bridge and the internal interface. Bind the IP address to the internal interface. Then STP and the application will use different MAC addresses, and we will be fine.
